eng/pipelines/compliance.yml

```diff
@@ -0,0 +1,80 @@
+resources:
+  pipelines:
+    - pipeline: NuGet.Client-Compliance
+      source: NuGet.Client-Official
```

Comment:
I'm not 100% sure, but I think this is saying "call NuGet.Client-Official by the name NuGet.Client-Compliance in this pipeline" [source](https://docs.microsoft.com/en-us/azure/devops/pipelines/yaml-schema?view=azure-devops&tabs=schema%2Cparameter-schema#pipeline-resource)

Reply:
The [doc](https://docs.microsoft.com/en-us/azure/devops/pipelines/process/pipeline-triggers?view=azure-devops#configure-pipeline-resource-triggers) has following information. `NuGet.Client-Compliance` is just the name of the resource and `NuGet.Client-Official` is the pipeline that triggers `new yml` upon successful completion of `dev` build.

``` yaml
# this is being defined in NEW app-ci pipeline
resources:
  pipelines:
  - pipeline: securitylib   # Name of the pipeline resource
    source: security-lib-ci # Name of the pipeline referenced by the pipeline resource
    project: FabrikamProject # Required only if the source pipeline is in another project
    trigger: true # Run app-ci pipeline when any run of security-lib-ci completes

The above settings work when we want the app-ci pipeline to automatically run every time a new version of security-lib-ci is built.
```
I added a `dev` branch `trigger` to run after successful completion of `official` build.

Reply:
In the future, when you want to download the official build's artifacts to run different compliance tasks on it, you'll need to do something like:

```yaml
- task: DownloadPipelineArtifact@2
  source: specific
  pipeline: NuGet.Client-Compliance
  artifact: VS15
  path: $(Pipeline.Workspace)\VS15
```

You're giving the NuGet.Client-Official resource the name NuGet.Client-Compliance, which is a bit confusing because the current pipeline is the compliance pipeline, not the "linked" pipeline.

Reply:
I agree this is bit confusing. I have edited the resource name to `nugetclientofficial` in the latest commit to avoid confusion in the future.

`SYSTEM_DEFINITIONID` value should be passed to `pipeline` parameter in `DownloadPipelineArtifact` artifacts task. 

https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/utility/download-pipeline-artifact?view=azure-devops#how-can-i-find-the-id-of-the-pipeline-i-want-to-download-an-artifact-from